### PR TITLE
CSS fixes and dependency upgrades

### LINKED ISSUE #56: Development build not downleveling nested selectors?
I'm trying to replace sass with css + nested selectors.  Everything looks fine in my production build, but the development build isn't providing valid styles to the browser.  Looking at the browser's network tab, it seems the css that it's being served has nesting built into it.  [No browser currently supports nesting](https://caniuse.com/?search=css-nesting) so I was expecting it to be transpiled.

`parcel` command:
```
parcel serve ./public/index.html --dist-dir .parcel-dev --public-url / --port 3000
```
`.parcelrc`:
```json
{
  "extends": "@parcel/config-default",
  "transformers": {
    "*.css": ["@parcel/transformer-css-experimental"]
  },
  "optimizers": {
    "*.css": ["@parcel/optimizer-css"]
  }
}
```

`@parcel/transformer-css` config in `package.json`:
```json
  "@parcel/transformer-css": {
    "drafts": {
      "nesting": true
    }
  }
```

`browserslist` config in `package.json`:
```json
  "browserslist": {
    "production": [
      "last 3 chrome version",
      "last 2 firefox version",
      "last 1 safari version",
      "firefox esr"
    ],
    "development": [
      "last 1 chrome version",
      "last 1 firefox version",
      "last 1 safari version"
    ]
  }
```

Full project: (Link is to the PR making these changes to add css): https://github.com/nattthebear/ff12characterplanner/pull/28

### LINKED ISSUE #7543: Regular Expression Denial of Service in postcss used by uncss
@parcel/optimizer-htmlnano@2.1.1 uses uncss@0.17.3 which depends on postcss@7.0.39.

In versions of postcss prior to 8.2.13 the following DoS was found on the 7th of January: ​https://github.com/advisories/GHSA-566m-qj78-rww5

### LINKED ISSUE #7556: In Dev, background image URIs are missing enclosed quotes
# 🐛 bug report

In Development, css property background uri's are missing their enclosing " " causing chrome to report invalid property value.

## 🎛 Configuration (.babelrc, package.json, cli command)

No Babelrc in project root (never needed one before), using Parcel stock / provided / built-in
```js
{
  dev: parcel watch --no-content-hash --no-hmr ./src/index.js
  build: cross-env NODE_ENV=production parcel build --no-source-maps ./src/index.js
  
  Node : 16.13
  Parcel: 2.2.0 (was forced to upgrade from 2.0.1)
  Sass transformer : 2.2.0 (was forced to upgrade from 2.0.1)
  React: 17+
}
```

## 🤔 Expected Behavior

Property value output should be valid in inspector for background URI 

## 😯 Current Behavior

Currently causes an invalid prop type and no longer renders the image in the DOM

Screen Shot 2022-01-13 at 12 17 02 PM

## 🔦 Context

Spent two - three hours thinking I was doing something wrong syntax wise, package upgrade, etc. Even converted the scss file to a styled js file to make it show up. A soon as I added " data:... " to the inspected image, waalaa.

css => 
```
background-image: url("data:image/svg+xml,%3Csvg xmlns='http://www.w3.org/2000/svg' height='20' width='20' viewBox='0 0 20 20' aria-hidden='true' focusable='false' class='bay-dropdown-svg'%3E%3Cpath d='M4.516 7.548c0.436-0.446 1.043-0.481 1.576 0l3.908 3.747 3.908-3.747c0.533-0.481 1.141-0.446 1.574 0 0.436 0.445 0.408 1.197 0 1.615-0.406 0.418-4.695 4.502-4.695 4.502-0.217 0.223-0.502 0.335-0.787 0.335s-0.57-0.112-0.789-0.335c0 0-4.287-4.084-4.695-4.502s-0.436-1.17 0-1.615z'%3E%3C/path%3E%3C/svg%3E");
```

Should be noted I did not see this issue when doing a production build output.

When using the sass transformer, in the development mode, enclosing quotes for the data URI should not be stripped or converted. If so they should respect the browser standard.

## 🌍 Your Environment

Also, I was trying to use 2.0.1 on Parcel and Sass Transformer and was forced through build error I had to use 2.2.0. Was not a huge fan of that. Outside of that, I love what parcel does for me on the regular. Thanks guys!

```
@parcel/package-manager: Could not find module "@parcel/transformer-sass" satisfying ^2.2.0.

~/.../project-root/package.json:84:5
    83 |     "@babel/core": "^7.15.0",
  > 84 |     "@parcel/transformer-sass": "^2.0.1",
  >    |     ^^^^^^^^^^^^^^^^^^^^^^^^^^ Found this conflicting local requirement.
    85 |     "cross-env": "^7.0.3",
```

| Software         | Version(s) |
| ---------------- | ---------- |
| Parcel           | 2.2.0
| Node             | 16.13
| npm/Yarn         | Yarn 1.22.17
| Operating System | OSX
|Sass | 1.44.0

### PR DESCRIPTION
**Pull request by devongovett**
* Fixes #7556. This was caused by a change I made to make the old CSS transformer in line with `@parcel/css`, but it resulted in a bug. Now we preserve the quotes, and I added them to Parcel CSS as well to match. That way we don't need to deal with escaping. That will be handled by the minifier if needed.
* Fixes https://github.com/parcel-bundler/parcel-css/issues/56. Back in #4838 we removed the default browser targets for perf reasons with Babel at the time. But now that we are on SWC it shouldn't make much difference if we default to the latest browser versions. That way features that aren't supported anywhere yet like CSS nesting will get compiled.

Also updates a bunch of dependencies to fix security warnings and reduce the number of duplicates we have:

* Fixes #7543. This bumps htmlnano to 2.0.0, which removes uncss. That results in many very old dependencies being removed, such as jsdom and old postcss versions.
* Removes css-module-loader-core which is unmaintained, and switches to the copy inside postcss-modules. Also upgrades postcss-modules. This removes all old versions of PostCSS from the lock file.
* Bumps the Vue compiler to the latest as well, to get rid of old PostCSS dep.
* Bumps SVGR to v6
* Normalizes version of JSON5 to v2
* http-proxy-middleware v2
* And several other random bumps noted in the security audit

Hopefully these solve the security audit issues for now, and starts to reduce the number of dependencies we have.

### CODE REVIEW

**Comment by parcel-benchmark** [comment]
## Benchmark Results
<details><summary>Kitchen Sink ✅</summary><p>

#### Timings

| Description | Time | Difference |
| --- | --- | --- |
| Cold | 1.92s | -0.00ms |
| Cached | 320.00ms | -10.00ms |

#### Cold Bundles

*No bundle changes detected.*

#### Cached Bundles

| Bundle | Size | Difference | Time | Difference |
| --- | --- | --- | --- | --- |
| dist/legacy/parcel.7cdb0fad.webp | 102.94kb | +0.00b | 240.00ms | -303.00ms 🚀 |
| dist/legacy/parcel.7cdb0fad.webp | 102.94kb | +0.00b | 240.00ms | -304.00ms 🚀 |
| dist/modern/parcel.7cdb0fad.webp | 102.94kb | +0.00b | 241.00ms | -303.00ms 🚀 |

</p></details>

<details><summary>React HackerNews ✅</summary><p>

#### Timings

| Description | Time | Difference |
| --- | --- | --- |
| Cold | 9.54s | +186.00ms |
| Cached | 456.00ms | +14.00ms |

#### Cold Bundles

*No bundle changes detected.*

#### Cached Bundles

*No bundle changes detected.*

</p></details>

<details><summary>AtlasKit Editor ✅</summary><p>

#### Timings

| Description | Time | Difference |
| --- | --- | --- |
| Cold | 1.04m | +543.00ms |
| Cached | 1.50s | +11.00ms |

#### Cold Bundles

| Bundle | Size | Difference | Time | Difference |
| --- | --- | --- | --- | --- |
| dist/index.7b91daf8.js | 1.78mb | +8.00b ⚠️ | 17.68s | +306.00ms |
| dist/index.c8c11eb0.js | 693.84kb | -2.00b 🚀 | 46.99s | -24.00ms |

#### Cached Bundles

| Bundle | Size | Difference | Time | Difference |
| --- | --- | --- | --- | --- |
| dist/index.6713c57d.js | 1.78mb | +8.00b ⚠️ | 17.57s | -18.00ms |
| dist/index.c8c11eb0.js | 693.84kb | -2.00b 🚀 | 47.10s | -26.00ms |
| dist/editorView.124f5fc0.js | 594.92kb | +0.00b | 46.16s | +11.26s ⚠️ |
| dist/popup.64bc9a82.js | 209.67kb | +0.00b | 46.16s | +11.26s ⚠️ |
| dist/Toolbar.1af0e801.js | 107.23kb | +0.00b | 46.16s | +11.26s ⚠️ |
| dist/Modal.cd71eaf3.js | 45.33kb | +0.00b | 46.16s | +11.26s ⚠️ |
| dist/ui.5d3f7adc.js | 14.94kb | +0.00b | 46.16s | +11.25s ⚠️ |
| dist/smartMediaEditor.48c8cf63.js | 13.25kb | +0.00b | 46.16s | +11.26s ⚠️ |
| dist/dropzone.39132d0c.js | 12.16kb | +0.00b | 46.16s | +11.26s ⚠️ |
| dist/EmojiPickerComponent.0482d6c0.js | 3.73kb | +0.00b | 46.16s | +11.25s ⚠️ |
| dist/dropzone.55bef257.js | 3.29kb | +0.00b | 46.16s | +11.26s ⚠️ |
| dist/clipboard.df70240c.js | 2.93kb | +0.00b | 46.16s | +11.26s ⚠️ |
| dist/ResourcedEmojiComponent.667554b4.js | 2.12kb | +0.00b | 46.16s | +11.25s ⚠️ |
| dist/browser.4e039ed7.js | 1.69kb | +0.00b | 46.16s | +11.26s ⚠️ |
| dist/media-card-analytics-error-boundary.c718a9a7.js | 1.12kb | +0.00b | 46.16s | +11.26s ⚠️ |
| dist/media-picker-analytics-error-boundary.8b2547e5.js | 966.00b | +0.00b | 46.16s | +11.26s ⚠️ |
| dist/simpleHasher.fc0d6100.js | 643.00b | +0.00b | 46.16s | +11.25s ⚠️ |

</p></details>

<details><summary>Three.js ✅</summary><p>

#### Timings

| Description | Time | Difference |
| --- | --- | --- |
| Cold | 6.63s | +125.00ms |
| Cached | 382.00ms | -0.00ms |

#### Cold Bundles

*No bundle changes detected.*

#### Cached Bundles

*No bundle changes detected.*

</p></details>

**[Click here to view a detailed benchmark overview.](https://benchmark.parceljs.org/benchmark/321049010850759250)**

**Comment by devongovett** [comment]
Ended up reverting the babel upgrade because it caused a significant performance regression. Parcel's code compiled with the new Babel runs slower for some reason...

File: packages/optimizers/css/package.json [hunk 1]
```diff
@@ -17,10 +17,10 @@
   "source": "src/CSSOptimizer.js",
   "engines": {
     "node": ">= 12.0.0",
-    "parcel": "^2.2.0"
+    "parcel": "^2.1.0"
```
Comment (thewilkybarkid) [inline packages/optimizers/css/package.json]: Is this meant?

**Comment by mischnic** [comment]
AFAICT, the only change from the Babel upgrade (so the state before the "Revert babel upgrade" commit) is that public and private class properties are transpiled.

File: packages/optimizers/css/package.json (hunk 1 above)
Comment (devongovett) [reply packages/optimizers/css/package.json]: Oops!

**Comment by devongovett** [comment]
Oh that's weird! Why would Babel start transpiling class properties?

**Comment by mischnic** [comment]
Happens in the Babel REPL as well, they are transpiled with Node <=14

https://babeljs.io/repl#?browsers=&build=&builtIns=false&corejs=false&spec=false&loose=false&code_lz=MYGwhgzhAEAa0G8BQ1oE9oF5oCYDcK0AxAF5bQAsBAvkA&debug=false&forceAllTransforms=false&shippedProposals=false&circleciRepo=&evaluate=true&fileSize=false&timeTravel=false&sourceType=module&lineWrap=true&presets=env&prettier=false&targets=Node-14&version=7.16.9&externalPlugins=&assumptions=%7B%7D

Maybe because these version don't actually support them 100%?
Bildschirmfoto 2022-01-17 um 16 25 51

**Comment by devongovett** [comment]
Hmm well I think it was working for us before just fine. Plus, node 12 is almost out of LTS (in April) so I'm not too concerned. Better to keep the improved performance.

**Comment by mischnic** [comment]
preset-env 7.14.0 is the first version that transpiles them, it works fine with
```json
{
  "dependencies": {
    "@babel/cli": "^7.16.8",
    "@babel/core": "^7.16.7",
    "@babel/preset-env": "^7.13.0"
  }
}
```
So probably somehow caused by https://github.com/babel/babel/pull/13091

> Plus, node 12 is almost out of LTS (in April) so I'm not too concerned.

Bumping the target from Node 12 to Node 14 still transpiles them. It's only with 16 that they are left in

**Comment by devongovett** [comment]
It's weird because this still shows node 12: https://github.com/babel/babel/blob/main/packages/babel-compat-data/data/plugins.json#L25. But running with the debug option in preset env shows this:

```
Using plugins:
  proposal-class-static-block { node < 16.11 }
  proposal-private-property-in-object { node < 16.9 }
  proposal-class-properties { node < 14.6 }
  proposal-private-methods { node < 14.6 }
```

I thought maybe disabling class static block, etc. would help but it was still transpiled with:

```js
exclude: [
  '@babel/plugin-proposal-private-methods',
  '@babel/plugin-proposal-private-property-in-object',
  '@babel/plugin-proposal-class-static-block'
]
```

**Comment by devongovett** [comment]
Going to merge this for now so we can get the security upgrades out, and file an issue with Babel.

**Review by mischnic — APPROVED** [review]

**Comment by devongovett** [comment]
https://github.com/babel/babel/issues/14167

**Pull request merged**